File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/callbacks/ConnectCallback.java
package com.smartwalkie.voicepingsdk.callbacks;

import com.smartwalkie.voicepingsdk.exceptions.PingException;

/**
 * Interface to provide callback to connect action.
 */
public interface ConnectCallback {

    /**
     * Invoked after successfully connected to server.
     */
    void onConnected();

    /**
     * Invoked on failed connecting process.
     *
     * @param exception
     */
    void onFailed(PingException exception);
}

File: voiceping-sdk/src/main/java/com/smartwalkie/voicepingsdk/callback/ConnectCallback.kt
package com.smartwalkie.voicepingsdk.callback

import com.smartwalkie.voicepingsdk.exception.VoicePingException

/**
 * Interface to provide callback to connect action.
 */
interface ConnectCallback {
    /**
     * Invoked after successfully connected to server.
     */
    fun onConnected()

    /**
     * Invoked on failed connecting process.
     *
     * @param exception
     */
    fun onFailed(exception: VoicePingException)
}